From bdb22809412fa2a9d804adbc30343621e928b42d Mon Sep 17 00:00:00 2001
From: "sheng.liu" <sheng.liu@amlogic.com>
Date: Wed, 13 Apr 2022 14:22:09 +0800
Subject: [PATCH] gstreamer1.0-libav: adjust libav patches [1/1]

PD#SWPL-78450

Problem:
Patches are not unique and independent

Solution:
Modify patch 0001 to enable ffmpeg default
Remove patch 003, 006, 007, 008
Add new patch to fix SWPL-70094 issue
Rename paches

Verify:
AP222
---
 ext/libav/gstavcodecmap.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ext/libav/gstavcodecmap.c b/ext/libav/gstavcodecmap.c
index 3c18ee8..9df1e10 100755
--- a/ext/libav/gstavcodecmap.c
+++ b/ext/libav/gstavcodecmap.c
@@ -1286,7 +1286,9 @@ gst_ffmpeg_codecid_to_caps (enum AVCodecID codec_id,
     case AV_CODEC_ID_H264:
       caps =
           gst_ff_vid_caps_new (context, NULL, codec_id, encode, "video/x-h264",
-          "alignment", G_TYPE_STRING, "au", NULL);
+          "parsed", G_TYPE_BOOLEAN, "true",
+          "alignment", G_TYPE_STRING, "au", 
+          "stream-format", G_TYPE_STRING, "byte-stream", NULL);
       if (!encode) {
         GValue arr = { 0, };
         GValue item = { 0, };
-- 
2.25.1

